src/main/kotlin/build/buf/intellij/formatter/BufFormatterService.kt

```diff
@@ -58,14 +59,24 @@ class BufFormatterService : AsyncDocumentFormattingService() {
                         preserveNewlines = true,
                     )
                 }
-                if (output.firstOrNull()?.startsWith("unknown command") == true) {
+                if (result.exitCode == 0) {
+                    request.onTextReady(result.stdout.joinToString(""))
+                    return
+                }
+                if (result.stderr.startsWith("unknown command")) {
                     request.onError(
                         BufBundle.message("formatter.title"),
                         BufBundle.message("formatter.cli.version.not.supported"),
                     )
-                } else {
-                    request.onTextReady(output.joinToString(separator = ""))
+                    return
                 }
+                val stderrLimited = result.stderr.split(System.lineSeparator()).stream()
+                    .limit(10)
+                    .collect(Collectors.joining(System.lineSeparator()))
+                request.onError(
+                    BufBundle.message("formatter.title"),
+                    stderrLimited,
+                )
```

Comment:
I'm a bit unsure whether a popup (I'm assuming this is how the error will show) is good DX here. I expect that the IDE will already report syntax errors, and this could have odd side-effects with "reformat code on save" enabled.

On the other hand, it's very possible that the IDE doesn't report all errors that prevent a successful `buf format`. Considering that the current DX with "reformat code on save" is currently much worse, I think it's a good call to add the popup.

Reply:
I wasn't sure the level of reporting we want here. Any errors running `buf format` will be logged as a warning to idea.log.

This doesn't show as a modal popup but instead a notification at the bottom right (which auto hides after a delay).

Reply:
I did check a few other format implementations that shell out to external processes and they do report errors this way (with stderr).